src/Symfony/Component/Yaml/Parser.php

```diff
@@ -250,19 +189,29 @@ public function parse($value, $flags = 0)
                         $realCurrentLineNbKey = $this->getRealCurrentLineNb();
                         $value = $this->parseBlock($this->getRealCurrentLineNb() + 1, $this->getNextEmbedBlock(), $flags);
                         // Spec: Keys MUST be unique; first one wins.
-                        // But overwriting is allowed when a merge node is used in current block.
-                        if ($allowOverwrite || !isset($data[$key])) {
+                        if (!isset($data[$key])) {
                             $data[$key] = $value;
+                        } elseif ('<<' === $key) {
+                            if (isset($data['<<'][0])) {
+                                $data['<<'][] = $value;
+                            } else {
+                                $data['<<'] = array($data['<<'], $value);
+                            }
                         } else {
                             @trigger_error(sprintf('Duplicate key "%s" detected on line %d whilst parsing YAML. Silent handling of duplicate mapping keys in YAML is deprecated since version 3.2 and will throw \Symfony\Component\Yaml\Exception\ParseException in 4.0.', $key, $realCurrentLineNbKey + 1), E_USER_DEPRECATED);
                         }
                     }
                 } else {
                     $value = $this->parseValue($values['value'], $flags, $context);
                     // Spec: Keys MUST be unique; first one wins.
-                    // But overwriting is allowed when a merge node is used in current block.
-                    if ($allowOverwrite || !isset($data[$key])) {
+                    if (!isset($data[$key])) {
                         $data[$key] = $value;
+                    } elseif ('<<' === $key) {
+                        if (isset($data['<<'][0])) {
+                            $data['<<'][] = $value;
+                        } else {
+                            $data['<<'] = array($data['<<'], $value);
```

Comment:
This will break the case where `$data['<<']` already is an array (see https://3v4l.org/9Tb7d).